include/picongpu/particles/atomicPhysics/AtomicPhysics.kernel

```diff
@@ -0,0 +1,362 @@
+                // note: maybe rename variables later
+                RngFactoryInt rngFactoryInt;
+                RngFactoryFloat rngFactoryFloat;
+
+                AtomicPhysicsKernel(
+                    RngFactoryInt rngFactoryInt = RngFactoryInt{0},
+                    RngFactoryFloat rngFactoryFloat = RngFactoryFloat{0})
+                    : rngFactoryInt(rngFactoryInt)
+                    , rngFactoryFloat(rngFactoryFloat)
+                {
+                }
+
+                // template parameters are automatically deduced from input
+                template<
+                    typename T_Acc,
+                    typename T_ElectronBox,
+                    typename T_IonBox,
+                    typename T_Mapping,
+                    typename T_AtomicDataBox>
+                HDINLINE void operator()(
+                    T_Acc const& acc,
+                    T_ElectronBox electronBox,
+                    T_IonBox ionBox,
+                    T_Mapping mapper,
+                    T_AtomicDataBox const atomicDataBox,
+                    float_X const initialGridWidth, // unit: ATOMIC_UNIT_ENERGY
+                    float_X const relativeErrorTarget, // unit: 1/s /( 1/( m^3 * ATOMIC_UNIT_ENERGY ) )
+                    uint32_t const step) const
+                {
+                    // we assume 1d thread indices
+                    // thread index inside a block
+                    uint32_t const workerIdx = cupla::threadIdx(acc).x;
+
+                    // Create and initialize a histogram on shared memory
+                    constexpr uint32_t frameSize = pmacc::math::CT::volume<SuperCellSize>::type::value;
+
+                    // TODO: get this from ion species
+                    // hard coded for now
+                    using ConfigNumberDataType = picongpu::atomicPhysics::configNumberDataType;
+                    // number of levels modelled
+                    constexpr uint8_t numLevels = picongpu::atomicPhysics::numLevels;
+
+                    // conversion to/from atomic state index
+                    using ConfigNumber = picongpu::particles::atomicPhysics::stateRepresentation::
+                        ConfigNumber<ConfigNumberDataType, numLevels, atomicDataBox.getAtomicNumber()>;
+
+                    // atomic rate calculation methods
+                    using AtomicRate = picongpu::particles::atomicPhysics::
+                        AtomicRate<T_AtomicDataBox, ConfigNumberDataType, ConfigNumber, numLevels>;
+
+                    using Histogram = electronDistribution::histogram2::
+                        AdaptiveHistogram<T_maxNumBins, frameSize, T_AtomicDataBox>;
+
+
+                    // debug only, compare to hand calculated rate
+                    /*
+                    ConfigNumberDataType oldState = 1055u;
+                    auto levelVector = ConfigNumber::getLevelVector(oldState);
+                    std::cout << "oldState: { " << uint16_t(levelVector[0]) << ", "
```

Comment:
`std::cout` within a kernel is not allowed.

Reply:
please do not forget to remove the includes too if this output is not required.

Reply:
done